instrumentation/grizzly-2.3/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/grizzly/GrizzlyFilterchainServerTest.java

```diff
@@ -0,0 +1,262 @@
+    TCPNIOTransportBuilder transportBuilder =
+        TCPNIOTransportBuilder.newInstance().setOptimizedForMultiplexing(true);
+
+    transportBuilder.setTcpNoDelay(true);
+    transportBuilder.setKeepAlive(false);
+    transportBuilder.setReuseAddress(true);
+    transportBuilder.setServerConnectionBackLog(50);
+    transportBuilder.setServerSocketSoTimeout(80000);
+
+    transport = transportBuilder.build();
+    transport.setProcessor(filterChain);
+  }
+
+  @SuppressWarnings(
+      "deprecation") // Constructor has been deprecated in favor of builder pattern promised to be
+  // released in a next major release version, but it is not yet available.
+  private static FilterChain setUpFilterChain() {
+    return FilterChainBuilder.stateless()
+        .add(createTransportFilter())
+        .add(createIdleTimeoutFilter())
+        .add(new HttpServerFilter())
+        .add(new LastFilter())
+        .build();
+  }
+
+  private static TransportFilter createTransportFilter() {
+    return new TransportFilter();
+  }
+
+  private static IdleTimeoutFilter createIdleTimeoutFilter() {
+    return new IdleTimeoutFilter(
+        new DelayedExecutor(Executors.newCachedThreadPool()), 80000, MILLISECONDS);
+  }
+
+  private static class LastFilter extends BaseFilter {
+    @Override
+    public NextAction handleRead(FilterChainContext ctx) throws IOException {
+      if (ctx.getMessage() instanceof HttpContent) {
+        HttpContent httpContent = ctx.getMessage();
+        HttpHeader httpHeader = httpContent.getHttpHeader();
+        if (httpHeader instanceof HttpRequestPacket) {
+          HttpRequestPacket request = (HttpRequestPacket) httpContent.getHttpHeader();
+          ResponseParameters responseParameters = buildResponse(request);
+          HttpResponsePacket.Builder builder =
+              HttpResponsePacket.builder(request)
+                  .status(responseParameters.getStatus())
+                  .header(
+                      "Content-Length",
+                      String.valueOf(responseParameters.getResponseBody().length));
+          responseParameters.fillHeaders(builder);
+          HttpResponsePacket responsePacket = builder.build();
+          controller(
+              responseParameters.getEndpoint(),
+              () -> {
+                responseParameters.execute();
+                ctx.write(
+                    HttpContent.builder(responsePacket)
+                        .content(wrap(ctx.getMemoryManager(), responseParameters.getResponseBody()))
+                        .build());
+                return responsePacket;
+              });
+        }
+      }
+      return ctx.getStopAction();
+    }
+
+    private static ResponseParameters buildResponse(HttpRequestPacket request) {
+      String uri = request.getRequestURI();
+      Map<String, String> headers = new HashMap<>();
+
+      ServerEndpoint endpoint;
+      Runnable closure = null;
+      switch (uri) {
```

Comment:
A more common pattern seems to be https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/f26a937ef4852fa05362694776da43d6d0fcb0c3/instrumentation/tomcat/tomcat-10.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/tomcat/v10_0/TestServlet.java#L22 to use `ServerEndpoint.forPath(path)`